pages/slider/basic.page.tsx

```diff
@@ -0,0 +1,259 @@
+                inputMode="numeric"
+                invalid={error2}
+                value={`${value2}`}
+                onChange={({ detail }) => {
+                  setValue2(Number(detail.value));
+                }}
+              />
+            </div>
+            <Box>Units</Box>
+          </SpaceBetween>
+        </div>
+      </FormField>
+
+      <FormField label="Slider with select">
+        <Grid gridDefinition={[{ colspan: { default: 4, xs: 2 } }]}>
+          <Select
+            selectedOption={minValue}
+            onChange={({ detail }) => setMinValue(detail.selectedOption)}
+            options={rangeOptions}
+          />
+        </Grid>
+        <Slider
+          value={Number(minValue.value)}
+          onChange={({ detail }) => {
+            setMinValue({ value: `${detail.value}` });
+          }}
+          min={5}
+          max={25}
+          step={5}
+          tickMarks={true}
+        />
+      </FormField>
+      <FormField label="Stepped slider">
+        <Slider
+          value={value3}
+          onChange={({ detail }) => {
+            setValue3(detail.value);
+          }}
+          valueFormatter={value => `${value} is the very long and precise value at this point.`}
+          min={0}
+          max={100}
+          step={10}
+          referenceValues={[30, 50, 55, 60, 75]}
+          tickMarks={true}
+        />
+      </FormField>
+      <FormField label="Where do you fall on the DT scale?">
+        <Slider
+          value={value4}
+          onChange={({ detail }) => {
+            setValue4(detail.value);
+          }}
+          valueFormatter={value => DTWordOptions.find(item => item.value === value)?.label || ''}
+          tickMarks={true}
+          min={0}
+          max={10}
+          step={2}
+          referenceValues={[2, 5, 8]}
+        />
+      </FormField>
+      <FormField label="How satisfied are you by your results?">
+        <Slider
+          value={value5}
+          onChange={({ detail }) => {
+            setValue5(detail.value);
+          }}
+          step={1}
+          min={0}
+          max={4}
+          valueFormatter={value => satisfactionScaleOptions.find(item => item.value === value)?.label || ''}
+          referenceValues={[1, 2, 3]}
+          tickMarks={true}
+        />
+      </FormField>
+      <FormField label="How hot do you like your salsa?">
+        <Slider
+          value={value6}
+          onChange={({ detail }) => {
+            setValue6(detail.value);
+          }}
+          step={0.1}
+          min={0}
+          max={1}
+          valueFormatter={value => `${value} is the value`}
+        />
+      </FormField>
+      <FormField label="How hot do you like your salsa?">
+        <Slider
+          value={value7}
+          onChange={({ detail }) => {
+            setValue7(detail.value);
+          }}
+          min={0}
+          max={100}
+          referenceValues={[7, 8, 9, 10, 25, 37, 39, 52, 53, 59, 67, 68, 93]}
```

Comment:
Happy to explain this one to you offline - went back and forth on the calculations here, but TL;DR - each label gets a set amount of space even if it can technically take up more.